crates/sui-graphql-rpc/src/test_infra/cluster.rs

```diff
@@ -190,3 +189,93 @@ pub async fn start_test_indexer(
     });
     (store, handle)
 }
+
+pub async fn simulator_commands_test_impl() {
+    let test_file_path = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("src")
+        .join("test_infra")
+        .join("data")
+        .join("example.move");
+
+    let output_file_path = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("src")
+        .join("test_infra")
+        .join("data")
+        .join("example.exp");
+
+    // Read the file into a string
+    let file_contents = std::fs::read_to_string(&test_file_path).unwrap();
+
+    let (output, adapter) = move_transactional_test_runner::framework::handle_actual_output::<
+        sui_transactional_test_runner::test_adapter::SuiTestAdapter,
+    >(
+        &test_file_path,
+        Some(&*sui_transactional_test_runner::test_adapter::PRE_COMPILED),
+    )
+    .await
+    .unwrap();
+
+    // Read the file into a string
+    let output_file_contents = match std::fs::read_to_string(&output_file_path) {
+        Ok(contents) => contents,
+        Err(_) => {
+            // If the file doesn't exist, create it
+            std::fs::write(&output_file_path, output.clone()).unwrap();
```

Comment:
Not sure I understand why this needs to be written to the output. It does not seem that the contents of this file is used in any way. Is it more for the user to manually verify the file for the errors when a test case has failed?

Reply:
> Is it more for the user to manually verify the file for the errors when a test case has failed?
Yes. This is essentially a snapshot test to make sure the output has not deviated between runs. You can imagine us using it to ensure graphql changes dont deviate.
Secondly, that file is an example for folks to see the behavior in action.